### PR TITLE
ci: .github/workflows: add codespell pipeline

### PR DESCRIPTION
**Pull request by invidian**
**Description of your changes:**

As a follow up to #6563 which fixed all the typos found in the
repository, to avoid adding more obvious typos.

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: .github/workflows/codespell.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  codespell:
+    name: codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: codespell
+        uses: codespell-project/actions-codespell@master
+        with:
+          skip: .git,*.png,*.jpg
+          ignore_words_list: aks,keyserver,atleast,dne,ser,ist,files',ba,dum,iam,te
```
Comment (leseb) [inline .github/workflows/codespell.yaml:21]: Do you have to increment this every time something new is not recognized?

Comment (invidian) [reply .github/workflows/codespell.yaml:21]: Yes.

File: .github/workflows/codespell.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  codespell:
+    name: codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: codespell
+        uses: codespell-project/actions-codespell@master
+        with:
+          skip: .git,*.png,*.jpg
```
Comment (satoru-takeuchi) [inline .github/workflows/codespell.yaml:20]: Please also add *.svg which is used for Rook's logo.

**Review by satoru-takeuchi — CHANGES_REQUESTED** [review]

File: .github/workflows/codespell.yaml (hunk 1 above)
Comment (satoru-takeuchi) [reply .github/workflows/codespell.yaml:21]: We would be able to use `ignore_words_files` here. However, I think listing words here is enough so far.

Comment (invidian) [reply .github/workflows/codespell.yaml:21]: In next release of codespell there will be config file available, which then will be preferable way of defining those I think.

File: .github/workflows/codespell.yaml (hunk 2 above)
Comment (invidian) [reply .github/workflows/codespell.yaml:20]: Well, they don't trigger errors right now, but sure :+1:

**Review by satoru-takeuchi — APPROVED** [review]

File: .github/workflows/codespell.yaml (hunk 1 above)
Comment (travisn) [reply .github/workflows/codespell.yaml:21]: - Do we need all these in the ignore list? Seems like we could fix some of them like "at least" or reword `files'` so we don't need the apostrophe. 
- Are these the only ignore words we need for a clean run? I'm surprised how short the list is.

Comment (invidian) [reply .github/workflows/codespell.yaml:21]: >Do we need all these in the ignore list? Seems like we could fix some of them like "at least" or reword files' so we don't need the apostrophe.

`atleast` is a method name. There is also `IsAtLeast`, so I think it's reasonable to ignore that. `files'` could be reworded, but I didn't want to increase the scope of the PR. 

>Are these the only ignore words we need for a clean run? I'm surprised how short the list is.

Well, workflows are passing here, soo yes. Do note, that this PR is a follow up to #6563, where I fixed all fixable typos :smile:

Comment (travisn) [reply .github/workflows/codespell.yaml:21]: Sounds good, and thanks for fixing all these issues! It's always a good feeling to improve the code hygiene.

**Review by leseb — APPROVED** [review]

**Pull request merged**